Sources/Networking/HTTPClient/SimpleNetworkServiceType.swift

```diff
@@ -0,0 +1,35 @@
+//
+//  Copyright RevenueCat Inc. All Rights Reserved.
+//
+//  Licensed under the MIT License (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      https://opensource.org/licenses/MIT
+//
+//  SimpleNetworkServiceType.swift
+//
+//  Created by Jacob Zivan Rakidzich on 8/12/25.
+
+import Foundation
+
+/// A protocol representing a simple network service
+@_spi(Internal) public protocol SimpleNetworkServiceType {
+    /// Fetch data from the network
+    /// - Parameter url: The URL to fetch data from
+    /// - Returns: Data upon success
+    func data(from url: URL) async throws -> Data
+}
+
+@_spi(Internal) extension URLSession: SimpleNetworkServiceType {
```

Comment:
Do these need to be `@_spi`? 🤔 Or will these only be used internally only in `RevenueCat` for now? It _seems_ like these mainly be used by the `FileRepository` in the short term. If so, let's remove `@_spi` and then just added it back whenever its needed.

But let me know if I'm missing an immediate use case 😅

Reply:
See other reply
https://github.com/RevenueCat/purchases-ios/pull/5477#discussion_r2284899741